src/csharp/Microsoft.Spark.Worker/Processor/BroadcastVariableProcessor.cs

```diff
@@ -55,17 +57,32 @@ internal BroadcastVariables Process(Stream stream)
                     else
                     {
                         string path = SerDe.ReadString(stream);
-                        // TODO: Register new broadcast variable.
+                        using FileStream fStream = File.Open(path, FileMode.Open, FileAccess.Read);
+                        formatter.Binder = new DeserializationBinder();
+                        var value = formatter.Deserialize(fStream);
+                        BroadcastRegistry.s_registry.TryAdd(bid, value);
                     }
                 }
                 else
                 {
                     bid = -bid - 1;
-                    // TODO: Remove registered broadcast variable.
+                    BroadcastRegistry.s_registry.TryRemove(bid, out _);
+                    BroadcastRegistry.s_listBroadcastVariables.Remove((int)bid);
                 }
             }
-
             return broadcastVars;
         }
     }
+
+    /// <summary>
+    /// Function that loads the application assembly and returns the Type of the broadcast
+    /// variable.
+    /// </summary>
+    sealed class DeserializationBinder : SerializationBinder
+    {
+        public override Type BindToType(string assemblyName, string typeName)
```

Comment:
Do you know if it's possible to initialize the following at the worker level so that this is not needed?
https://github.com/dotnet/spark/blob/40c9febd54f52464148ed69acc0371f95ffa9e28/src/csharp/Microsoft.Spark.Worker/Processor/CommandProcessor.cs#L24-L33

Reply:
@imback82  Moved this logic to `PayloadProcessor.cs` to remove duplication at `CommandProcessor` and `BroadcastVariableProcessor`. Please let me know if this makes sense. Thanks!